### ISSUE TITLE
profiles expect directory to exist

### ISSUE THREAD

**Issue by xer0x** [joyent/node-triton#65]
It looks like profiles expect ~/.triton/profiles.d to already exist:

```
$ triton profile -a
fs.js:761
  return binding.readdir(pathModule._makeLong(path));
                 ^
Error: ENOENT, no such file or directory '/Users/drew/.triton/profiles.d'
    at Error (native)
    at Object.fs.readdirSync (fs.js:761:18)
    at Object.loadAllProfiles (/Users/drew/.nvm/versions/node/v0.12.7/lib/node_modules/triton/lib/config.js:302:20)
    at Object.gatherDataInteractive [as func] (/Users/drew/.nvm/versions/node/v0.12.7/lib/node_modules/triton/lib/do_profile.js:341:39)
    at Immediate._onImmediate (/Users/drew/.nvm/versions/node/v0.12.7/lib/node_modules/triton/node_modules/vasync/lib/vasync.js:213:20)
    at processImmediate [as _immediateCallback] (timers.js:367:17)
```

**Comment by trentm** [comment]
Likewise, if one doesn't have relevant envvars setup (a reasonable state for people just starting with Triton), then the "env" profile blows up:

```
$ unset SDC_ACCOUNT
$ triton -v profiles
{"name":"triton","hostname":"danger0.local","pid":71793,"level":10,"err":{"message":"profile.account (string) is required","name":"VError","stack":"ConfigError: profile.account (string) is required\n    at ConfigError._TritonBaseVError (/Users/trentm/joy/node-triton/lib/errors.js:54:12)\n    at new ConfigError (/Users/trentm/joy/node-triton/lib/errors.js:138:23)\n    at validateProfile (/Users/trentm/joy/node-triton/lib/config.js:197:15)\n    at _loadEnvProfile (/Users/trentm/joy/node-triton/lib/config.js:246:5)\n    at Object.loadAllProfiles (/Users/trentm/joy/node-triton/lib/config.js:298:9)\n    at _listProfiles (/Users/trentm/joy/node-triton/lib/do_profiles.js:31:31)\n    at CLI.do_profiles (/Users/trentm/joy/node-triton/lib/do_profiles.js:79:5)\n    at CLI.dispatch (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:744:13)\n    at /Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:434:18\n    at CLI.init (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:555:5)","code":"Config"},"subcmd":"profiles","msg":"cli fini","time":"2015-12-29T21:43:28.816Z","src":{"file":"/Users/trentm/joy/node-triton/lib/cli.js","line":272,"func":"fini"},"v":0}
triton profiles: error (Config): ConfigError: profile.account (string) is required
    at ConfigError._TritonBaseVError (/Users/trentm/joy/node-triton/lib/errors.js:54:12)
    at new ConfigError (/Users/trentm/joy/node-triton/lib/errors.js:138:23)
    at validateProfile (/Users/trentm/joy/node-triton/lib/config.js:197:15)
    at _loadEnvProfile (/Users/trentm/joy/node-triton/lib/config.js:246:5)
    at Object.loadAllProfiles (/Users/trentm/joy/node-triton/lib/config.js:298:9)
    at _listProfiles (/Users/trentm/joy/node-triton/lib/do_profiles.js:31:31)
    at CLI.do_profiles (/Users/trentm/joy/node-triton/lib/do_profiles.js:79:5)
    at CLI.dispatch (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:744:13)
    at /Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:434:18
    at CLI.init (/Users/trentm/joy/node-triton/node_modules/cmdln/lib/cmdln.js:555:5)
```

**Comment by trentm** [comment]
Also adding a profile fails if that dir doesn't already exist:

```
$ triton profile -a
A profile name. A short string to identify a CloudAPI endpoint to the
`triton` CLI.
name: foo

The CloudAPI endpoint URL.
url: https://us-sw-1.api.joyent.com

Your account login name.
account: trent.mick

The fingerprint of the SSH key you have registered for your account. You may
enter a local path to a public or private key to have the fingerprint
calculated for you.
keyId: ~/.ssh/id_rsa
Fingerprint: de:e7:73:9a:b0:91:31:3e:72:8d:9f:62:ca:58:a2:ec

triton profile: error (ENOENT): ENOENT, no such file or directory '/Users/trentm/.triton/profiles.d/foo.json'
```

**Comment by trentm** [comment]
Also `triton profiles` borks if don't have a current profile (which defaults to 'env', so if don't have envvars set):

```
$ triton profiles
triton profiles: error: opts.profile (object) is required
```

**Comment by trentm** [comment]
showing it working with coming patch:

```
$ triton profiles
NAME  CURR  ACCOUNT  USER  URL
[14:29:36 trentm@danger0:~/joy/node-triton (master)]
$ triton profile -a
A profile name. A short string to identify a CloudAPI endpoint to the
`triton` CLI.
name: foo

The CloudAPI endpoint URL.
url: https://us-sw-1.api.joyent.com

Your account login name.
account: trent.mick

The fingerprint of the SSH key you have registered for your account. You may
enter a local path to a public or private key to have the fingerprint
calculated for you.
keyId: ~/.ssh/id_rsa
Fingerprint: de:e7:73:9a:b0:91:31:3e:72:8d:9f:62:ca:58:a2:ec

Saved profile "foo"
Set "foo" as current profile (because it is your only profile)
[14:29:48 trentm@danger0:~/joy/node-triton (master)]
$ triton profiles
NAME  CURR  ACCOUNT     USER  URL
foo   *     trent.mick  -     https://us-sw-1.api.joyent.com
[14:30:14 trentm@danger0:~/joy/node-triton (master)]
$ triton profile
name: foo
account: trent.mick
curr: true
keyId: de:e7:73:9a:b0:91:31:3e:72:8d:9f:62:ca:58:a2:ec
url: https://us-sw-1.api.joyent.com
```

**Comment by trentm** [comment]
This is fixed in triton@3.4.2
